client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -3,156 +3,74 @@
  */
 import { __ as NO__ } from '@wordpress/i18n';
 import { Button } from '@wordpress/components';
-import { isEmpty, map } from 'lodash';
-import { useDispatch, useSelect } from '@wordpress/data';
-import React, { useCallback, useState } from 'react';
-import classNames from 'classnames';
+import { useSelect } from '@wordpress/data';
+import React, { useState } from 'react';
+
 /**
  * Internal dependencies
  */
-import { SiteType, isFilledFormValue } from '../stores/onboard/types';
+import { isFilledFormValue } from '../stores/onboard/types';
 import { STORE_KEY } from '../stores/onboard';
+import Question from './question';
 import VerticalSelect from './vertical-select';
+import SiteTypeSelect, { siteTypeOptions } from './site-type-select';
+import SiteTitle from './site-title';
 import './style.scss';
 
-const siteTypeOptions: Record< SiteType, string > = {
-	[ SiteType.BLOG ]: NO__( 'with a blog' ),
-	[ SiteType.BUSINESS ]: NO__( 'for a business' ),
-	[ SiteType.PORTFOLIO ]: NO__( 'to share a portfolio' ),
-	[ SiteType.STORE ]: NO__( 'for a store' ),
-};
-
 export default function OnboardingEdit() {
-	const [ lastSelectedQuestion, setLastSelectedQuestion ] = useState( null );
+	const [ activeStep, setActiveStep ] = useState( 0 );
 
-	const { siteTitle, siteType, siteVertical } = useSelect( select =>
+	const { siteType, siteVertical, siteTitle } = useSelect( select =>
 		select( STORE_KEY ).getState()
 	);
 
-	const { setSiteType, setSiteTitle } = useDispatch( STORE_KEY );
-
-	const updateTitle = useCallback(
-		( e: React.ChangeEvent< HTMLInputElement > ) => setSiteTitle( e.target.value ),
-		[ setSiteTitle ]
-	);
-
-	const updateSiteType = useCallback(
-		( e: React.ChangeEvent< HTMLInputElement > ) => setSiteType( e.target.value as SiteType ),
-		[ setSiteType ]
-	);
-
-	const selectSiteType = e => {
-		updateSiteType( e );
-		setLastSelectedQuestion( null );
-	};
-
-	const openSiteType = () => {
-		setLastSelectedQuestion( 'siteType' );
-	};
-
-	const selectSiteVertical = () => {
-		setLastSelectedQuestion( 'siteVertical' );
-	};
-
-	const selectSiteTitle = () => {
-		setLastSelectedQuestion( 'siteTitle' );
-	};
-
-	const isSelected = questionType => {
-		if ( lastSelectedQuestion ) {
-			return questionType === lastSelectedQuestion;
-		} else if ( isEmpty( siteType ) ) {
-			return questionType === 'siteType';
-		} else if ( isEmpty( siteVertical ) ) {
-			return questionType === 'siteVertical';
-		} else if ( questionType === 'siteTitle' ) {
-			return true;
-		}
-		return false;
-	};
+	const handleNext = () => setActiveStep( activeStep + 1 );
+
+	const steps = [
+		{
+			value: siteType,
+			label: isFilledFormValue( siteType )
+				? NO__( 'I want to create a website ' )
+				: NO__( 'Setting up a website ' ),
+			displayValue: isFilledFormValue( siteType ) ? siteTypeOptions[ siteType ] : '',
```

Comment:
This information is tightly coupled to the `SiteTypeSelect` component and shouldn't live outside of it -- we're basically splitting the source of truth for infomation pertaining to this step, which is a rather confusing pattern.

If we want to use a generic `Question` component to be re-used for each question, the pattern should be that each of them uses the `Question` component via composition, not the other way round.

Reply:
I _think_ that some of this might be coming from how we're iterating over `steps` here, and render a `Question` component for each. Let's ditch that, and use `React.Children.map()` instead, and just wrap the steps in `OnboardingEdit` (or possibly an intermediary component). Something like:

```js
<StepperWizard>
    <SiteTypeSelect />
    <VerticalSelect />
    <SiteTitle />
```